File: app/src/main/java/com/komica/reader/KomicaReaderApplication.java
package com.komica.reader;

import android.app.Application;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;
import androidx.lifecycle.ProcessLifecycleOwner;

public class KomicaReaderApplication extends Application {
    private static boolean isAppInForeground = false;

    @Override
    public void onCreate() {
        super.onCreate();
        ProcessLifecycleOwner.get().getLifecycle().addObserver(new LifecycleObserver() {
            @OnLifecycleEvent(Lifecycle.Event.ON_START)
            public void onAppForeground() {
                isAppInForeground = true;
            }

            @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
            public void onAppInBackground() {
                isAppInForeground = false;
            }
        });
    }

    public static boolean isAppInForeground() {
        return isAppInForeground;
    }
}

File: app/src/main/java/com/komica/reader/KomicaReaderApplication.kt
package com.komica.reader

import android.app.Application
import androidx.lifecycle.DefaultLifecycleObserver
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.ProcessLifecycleOwner
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class KomicaReaderApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        ProcessLifecycleOwner.get().lifecycle.addObserver(object : DefaultLifecycleObserver {
            override fun onStart(owner: LifecycleOwner) {
                isAppInForeground = true
            }

            override fun onStop(owner: LifecycleOwner) {
                isAppInForeground = false
            }
        })
    }

    companion object {
        var isAppInForeground: Boolean = false
            private set
    }
}